src/pgen/standing_accretion_shock.cpp

```diff
@@ -0,0 +1,248 @@
+  const int iprs = imap[fluid_prim::pressure].first;
+  const int itmp = imap[fluid_prim::temperature].first;
+  const int igm1 = imap[fluid_prim::gamma1].first;
+
+  const std::string eos_type = pin->GetString("eos", "type");
+  PARTHENON_REQUIRE_THROWS(
+      eos_type == "StellarCollapse",
+      "Standing Accretion Shock setup only works with StellarCollapse EOS");
+
+  Real Mdot = pin->GetOrAddReal("standing_accretion_shock", "Mdot", 0.2);
+  Real rShock = pin->GetOrAddReal("standing_accretion_shock", "rShock", 200);
+  const Real target_mach =
+      pin->GetOrAddReal("standing_accretion_shock", "target_mach", 100);
+
+  IndexRange ib = pmb->cellbounds.GetBoundsI(IndexDomain::entire);
+  IndexRange jb = pmb->cellbounds.GetBoundsJ(IndexDomain::entire);
+  IndexRange kb = pmb->cellbounds.GetBoundsK(IndexDomain::entire);
+
+  auto &coords = pmb->coords;
+  auto eos = pmb->packages.Get("eos")->Param<Microphysics::EOS::EOS>("d.EOS");
+  auto Tmin = pmb->packages.Get("eos")->Param<Real>("T_min");
+  auto Tmax = pmb->packages.Get("eos")->Param<Real>("T_max");
+
+  const Real a = pin->GetReal("geometry", "a");
+  auto bl = Geometry::BoyerLindquist(a);
+
+  auto &unit_conv =
+      pmb->packages.Get("phoebus")->Param<phoebus::UnitConversions>("unit_conv");
+
+  Mdot *= ((solar_mass * unit_conv.GetMassCGSToCode()) / unit_conv.GetTimeCGSToCode());
+  rShock *= (1.e5 * unit_conv.GetLengthCGSToCode());
+  const Real MPNS = 1.3 * solar_mass * unit_conv.GetMassCGSToCode();
+  const Real rs =
+      (2. * pc.g_newt * MPNS / (std::pow(pc.c, 2))) * unit_conv.GetLengthCGSToCode();
+
+  printf("rShock (code units) = %g \n", rShock);
+
+  auto geom = Geometry::GetCoordinateSystem(rc);
+
+  pmb->par_for(
+      "Phoebus::ProblemGenerator::StandingAccrectionShock", kb.s, kb.e, jb.s, jb.e, ib.s,
+      ib.e, KOKKOS_LAMBDA(const int k, const int j, const int i) {
+        Real x1 = coords.Xc<1>(k, j, i);
+        const Real x2 = coords.Xc<2>(k, j, i);
+        const Real x3 = coords.Xc<3>(k, j, i);
+
+        Real r = std::abs(x1);
+        const Real gamma = 4. / 3.;
+        const Real epsND = 0.003; // 2.7e16 ergs / g for M = 1.3Mpns
+
+        Real eos_lambda[2];
+        if (iye > 0) {
+          v(iye, k, j, i) = 0.5;
+          eos_lambda[0] = v(iye, k, j, i);
+        }
+
+        // postshock - 1
+        if (r < rShock) {
+
+          const Real lapse0 = geom.Lapse(CellLocation::Cent, k, j, rShock);
+          const Real vr0 = -1. * std::sqrt(lapse0 * lapse0 - 1.);
+          const Real W0 = 1. / sqrt(1. - std::pow(vr0, 2));
+          Real rho0 = Mdot / (4. * M_PI * std::pow(r, 2) * W0 * std::abs(vr0));
+
+          Real alphasq = 1. - (2. / r);
+          Real psi = alphasq * ((gamma - 1.) / gamma) * ((W0 - 1.) / W0);
+          Real vr1 = (vr0 + std::sqrt(vr0 * vr0 - 4. * psi)) / 2.;
+          Real rho1 = rho0 * W0 * (vr0 / vr1);
+
+          v(irho, k, j, i) = rho1;
+          v(ieng, k, j, i) = (W0 - 1. + epsND * (gamma - 1.)) / (gamma);
```

Comment:
`v(ieng, k, j, i)` is internal energy by volume and formula from Radice is specific. Need to multiply by rho.

Reply:
Need to solve Radice's shock equations for EOS without assuming gamma

Reply:
So the issue I have run into is needing to choose a variable that we know to solve the rest. I settled on using the equation for velocity in the preshock region assuming ideal gas: const Real vr1 = (vr0 + std::sqrt(vr0 * vr0 - 4. * psi)) / 2.;

Then the density follows from the Rankine–Hugoniot jump conditions (no ideal gas assumption).

Then I solve the remaining equations without ideal gas assumption using vr1,vr0,rho0,rho1.